cmd/goal/clerk.go

```diff
@@ -1096,22 +1095,15 @@ var dryrunCmd = &cobra.Command{
 			if uint64(txn.Lsig.Len()) > params.LogicSigMaxSize {
 				reportErrorf("program size too large: %d > %d", len(txn.Lsig.Logic), params.LogicSigMaxSize)
 			}
-			ep := logic.EvalParams{Txn: &txn, Proto: &params, GroupIndex: uint64(i), TxnGroup: txgroup}
-			err := logic.Check(txn.Lsig.Logic, ep)
+			ep := logic.NewEvalParams(txgroup, &params, nil)
+			err := logic.CheckSignature(i, ep)
 			if err != nil {
 				reportErrorf("program failed Check: %s", err)
 			}
-			sb := strings.Builder{}
-			ep = logic.EvalParams{
-				Txn:        &txn,
-				GroupIndex: uint64(i),
-				Proto:      &params,
-				Trace:      &sb,
-				TxnGroup:   txgroup,
-			}
-			pass, err := logic.Eval(txn.Lsig.Logic, ep)
+			ep.Trace = &strings.Builder{}
+			pass, err := logic.EvalSignature(i, ep)
 			// TODO: optionally include `inspect` output here?
-			fmt.Fprintf(os.Stdout, "tx[%d] trace:\n%s\n", i, sb.String())
+			fmt.Fprintf(os.Stdout, "tx[%d] trace:\n%s\n", i, ep.Trace.String())
```

Comment:
if the trace could contain any non-printable characters, I suggest you'd use `reportInfof` instead.
( to filter out control characters and other non-printable characters )

Reply:
I don't think so, since we print byteslices as as hex.